cli/resolver.rs

```diff
@@ -1035,138 +1039,205 @@ impl SloppyImportsResolver {
-          MediaType::JavaScript => {
-            if mode.is_types() {
-              vec![MediaType::TypeScript, MediaType::Tsx, MediaType::Dts]
-            } else {
-              vec![MediaType::TypeScript, MediaType::Tsx]
-            }
-          }
-          MediaType::Jsx => vec![MediaType::Tsx],
-          MediaType::Mjs => {
-            if mode.is_types() {
-              vec![MediaType::Mts, MediaType::Dmts, MediaType::Dts]
-            } else {
-              vec![MediaType::Mts]
-            }
-          }
-          MediaType::Cjs => {
+
+          if matches!(entry, Some(SloppyImportsFsEntry::Dir)) {
+            // try to resolve at the index file
             if mode.is_types() {
-              vec![MediaType::Cts, MediaType::Dcts, MediaType::Dts]
+              probe_paths.push((
+                path.join("index.ts"),
+                SloppyImportsResolutionReason::Directory,
+              ));
+
+              probe_paths.push((
+                path.join("index.mts"),
+                SloppyImportsResolutionReason::Directory,
+              ));
+              probe_paths.push((
+                path.join("index.d.ts"),
+                SloppyImportsResolutionReason::Directory,
+              ));
+              probe_paths.push((
+                path.join("index.d.mts"),
+                SloppyImportsResolutionReason::Directory,
+              ));
+              probe_paths.push((
+                path.join("index.js"),
+                SloppyImportsResolutionReason::Directory,
+              ));
+              probe_paths.push((
+                path.join("index.mjs"),
+                SloppyImportsResolutionReason::Directory,
+              ));
+              probe_paths.push((
+                path.join("index.tsx"),
+                SloppyImportsResolutionReason::Directory,
+              ));
+              probe_paths.push((
+                path.join("index.jsx"),
+                SloppyImportsResolutionReason::Directory,
+              ));
             } else {
-              vec![MediaType::Cts]
+              probe_paths.push((
+                path.join("index.ts"),
+                SloppyImportsResolutionReason::Directory,
+              ));
+              probe_paths.push((
+                path.join("index.mts"),
+                SloppyImportsResolutionReason::Directory,
+              ));
+              probe_paths.push((
+                path.join("index.tsx"),
+                SloppyImportsResolutionReason::Directory,
+              ));
+              probe_paths.push((
+                path.join("index.js"),
+                SloppyImportsResolutionReason::Directory,
+              ));
+              probe_paths.push((
+                path.join("index.mjs"),
+                SloppyImportsResolutionReason::Directory,
+              ));
+              probe_paths.push((
+                path.join("index.jsx"),
+                SloppyImportsResolutionReason::Directory,
+              ));
```

Comment:
We should probably eventually switch this to an iterator.

Reply:
Yup - this could all use a cleanup I agree